pkg/splunk/enterprise/workerpool_test.go

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) 2018-2021 Splunk Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enterprise
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	enterpriseApi "github.com/splunk/splunk-operator/pkg/apis/enterprise/v2"
+	splclient "github.com/splunk/splunk-operator/pkg/splunk/client"
+	splcommon "github.com/splunk/splunk-operator/pkg/splunk/common"
+	spltest "github.com/splunk/splunk-operator/pkg/splunk/test"
+	splutil "github.com/splunk/splunk-operator/pkg/splunk/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// initialize some data structures to be used in this file
+func init() {
+	appSrcName = "appSrc1"
+	scope = "local"
+	crName = "dummyStandalone"
+	randomTime = "1633975734"
+	storageClass = "gp2"
+
+	appNames = []string{"app1.tgz", "app2.tgz", "app3.tgz", "app4.tgz"}
+	objectHashes = []string{"abcd1111", "efgh2222", "ijkl3333", "wxyz4444"}
```

Comment:
Do we have a test case that cover an updated hash/app?

Reply:
I think that is covered as part of the `util` framework. We reach at this code if there are any new apps or updated apps and hence we download them.